shader/flatten.wgsl

```diff
@@ -204,89 +272,191 @@ fn round_up(x: f32) -> i32 {
     return i32(ceil(x));
 }
 
-@compute @workgroup_size(256)
-fn main(
-    @builtin(global_invocation_id) global_id: vec3<u32>,
-    @builtin(local_invocation_id) local_id: vec3<u32>,
-) {
-    let ix = global_id.x;
+struct PathTagData {
+    tag_byte: u32,
+    monoid: TagMonoid,
+}
+
+fn compute_tag_monoid(ix: u32) -> PathTagData {
     let tag_word = scene[config.pathtag_base + (ix >> 2u)];
-    pathdata_base = config.pathdata_base;
     let shift = (ix & 3u) * 8u;
     var tm = reduce_tag(tag_word & ((1u << shift) - 1u));
     // TODO: this can be a read buf overflow. Conditionalize by tag byte?
     tm = combine_tag_monoid(tag_monoids[ix >> 2u], tm);
     var tag_byte = (tag_word >> shift) & 0xffu;
+    return PathTagData(tag_byte, tm);
+}
 
-    let out = &path_bboxes[tm.path_ix];
-    let style_flags = scene[config.style_base + tm.style_ix];
-    // TODO: We assume all paths are fills at the moment. This is where we will extract the stroke
-    // vs fill state using STYLE_FLAGS_STYLE_BIT.
-    let draw_flags = select(DRAW_INFO_FLAGS_FILL_RULE_BIT, 0u, (style_flags & STYLE_FLAGS_FILL_BIT) == 0u);
-    if (tag_byte & PATH_TAG_PATH) != 0u {
-        (*out).draw_flags = draw_flags;
-        (*out).trans_ix = tm.trans_ix;
-    }
-    // Decode path data
-    let seg_type = tag_byte & PATH_TAG_SEG_TYPE;
-    if seg_type != 0u {
-        var p0: vec2<f32>;
-        var p1: vec2<f32>;
-        var p2: vec2<f32>;
-        var p3: vec2<f32>;
-        if (tag_byte & PATH_TAG_F32) != 0u {
-            p0 = read_f32_point(tm.pathseg_offset);
-            p1 = read_f32_point(tm.pathseg_offset + 2u);
-            if seg_type >= PATH_TAG_QUADTO {
-                p2 = read_f32_point(tm.pathseg_offset + 4u);
-                if seg_type == PATH_TAG_CUBICTO {
-                    p3 = read_f32_point(tm.pathseg_offset + 6u);
-                }
+struct CubicPoints {
+    p0: vec2f,
```

Comment:
`vec2f` is the shorthand for `vec2<f32>` in type declarations. `vec2` seems to work as a constructor expression where the element type gets inferred. I've been using `vec2f` in new type declarations and I tried to match `vec2(...)` in constructor expressions since that's what the existing code did. I'd be happy to standardize on `vec2f` though.